Update C changelog for C API 0.99.15 (#2027)
tskit-dev/tskit: c/CHANGELOG.rst

  tree sequence. This is then used to geerate an error if ``time_units`` is ``uncalibrated`` when
  using the branch lengths in statistics. (:user:`benjeffery`, :issue:`1644`, :pr:`1760`)

- Add the ``TSK_LOAD_SKIP_TABLES`` option to load just the top-level information from a
  file. Also add the ``TSK_CMP_IGNORE_TABLES`` option to compare only the top-level
  information in two table collections. (:user:`clwgg`, :pr:`1882`, :issue:`1854`).

- Add reference sequence.
  (:user:`jeromekelleher`, :user:`benjeffery`, :issue:`146`, :pr:`1911`, :pr:`1944`, :pr:`1911`)

- Add the ``TSK_LOAD_SKIP_REFERENCE_SEQUENCE`` option to load a table collection
  without the reference sequence. Also add the TSK_CMP_IGNORE_REFERENCE_SEQUENCE
  option to compare two table collections without comparing their reference
  sequence. (:user:`clwgg`, :pr:`2019`, :issue:`1971`).

- Add a "virtual root" to ``Tree`` arrays such as ``left_sib``, ``right_child`` etc.

>   The root inserted at the end of the array, has all real roots as it's children,

Reviewer: ```suggestion
  The virtual root is appended to each array, has all real roots as its children,
```

Author: Done

  but is not the parent of any node. Simplifies traversal algorithms.
  (:user:`jeromekelleher`, :issue:`1691`, :pr:`1704`)

- Add ``num_edges`` to ``tsk_tree_t`` to count the edges that define the topology of

>   the tree. (:user:`jeromekelleher`, :pr:`1704`)

Reviewer: ```suggestion
  the tree. (:user:`jeromekelleher`, :pr:`1704`)
  
- Add the ``tsk_tree_get_size_bound`` function which returns an upper bound on the number of nodes reachable from 
  the roots of a tree. Useful for tree stack allocations (:user:`jeromekelleher`, :pr:`1704`).
```

Author: Done
